### PR TITLE
New documentation: Objects and Resource Nodes #1040

### PR DESCRIPTION
**Pull request by manez**
**GitHub Issue**: https://github.com/Islandora/documentation/issues/1040

[Original document](https://docs.google.com/document/d/1fKynWjP7d-g16Eh0AMBvWXPnasJie-xEzmzajukhUww/edit?usp=sharing)

# What does this Pull Request do?

Takes the conceptual parts of @alexkent0 's document introducing Islandora 8 from an islandora 7 end-user perspective and makes a new page for our docs. This is the source of the term "resource node."

The original document has a lot of additional content that I left out, as it's covered elsewhere in our documentation, but these conceptual bits are unique. 

All done with Alex's permission and approval.

# What's new?
A page of conceptual documentation and a screenshot. 

# How should this be tested?

Preview, read through, give opinions 😄 

# Additional Notes:
Any additional information that you think would be helpful when reviewing this PR.

# Interested parties
@Islandora/8-x-committers and @alexkent0

### CODE REVIEW

File: docs/user-documentation/objects_to_resource_nodes.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
+ # From Objects to Resource Nodes: Shifting Concepts from Islandora 7 to 8
+
+This document attempts to show the shift in how objects work from Islandora version 7 to version 8, from a non-developer perspective.
+
+## Building resources in Islandora 8
+
+Islandora 8 represents a fundamental shift forward from Islandora 7. Islandora 8 turns what are known as traditional Islandora objects into Drupal nodes. It is possible to manipulate all aspects of objects through the Drupal user interface. Non-developers can configure and set up a myriad of content and functions in Islandora that enhance the discovery of and support access to objects in the repository. Improving the discovery of and access to objects is a significant theme in Islandora. 
+
+Another difference from Islandora 7 to 8 is with Islandora objects. In Islandora 7, _objects_ (video files, audio files, PDFs, etc.) were loaded through the user interface, and Datastreams were generated automatically. These consisted of access and display copies, the metadata, OCH/HOCR, technical metadata, and more. All of these Datastreams were directly connected to the object and accessed through the admin interface.
```
Comment (librarychik) [inline docs/user-documentation/objects_to_resource_nodes.md]:
```suggestion
Another difference between Islandora 7 and 8 are Islandora objects. In Islandora 7, _objects_ (video files, audio files, PDFs, etc.) are loaded through the user interface, and Datastreams are generated automatically. These consist of access and display copies, the metadata, OCH/HOCR, technical metadata, and more. All of these Datastreams are directly connected to the object and accessed through the admin interface. 
```

File: docs/user-documentation/objects_to_resource_nodes.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+In Islandora 8, the traditional Islandora 7 objects (video files, audio files, etc.) are now _Drupal nodes_. Islandora object nodes are a special kind of Drupal node, distinct from nodes that exist for other content such as a blog post, an article, a webpage (like the About page on a site), and others. These Islandora objects are still loaded through the interface and described with the data entry form, and derivatives are still generated. However, the Datastreams are no longer connected to the original object in the same immutable way. Each of these Datastreams can be manipulated through Drupal by non-developers. A myriad of different ways to view this metadata and information related to the objects can be created and set up. Doing so does take strong knowledge of Drupal 8. But this essentially means that there are a lot more ways to view the metadata and access the related objects in Islandora 8.
```
Comment (librarychik) [inline docs/user-documentation/objects_to_resource_nodes.md]:
```suggestion
In Islandora 8, the traditional Islandora 7 objects (video files, audio files, etc.) are now _Drupal nodes_. Islandora object nodes are a special kind of Drupal node, distinct from nodes that exist for other content types such as a blog post, an article, a page (like the About page on a site), and others. These Islandora objects are still loaded through the interface and described with the data entry form and derivatives are still generated. However, the Datastreams are no longer connected to the original object in the same immutable way. Each of these Datastreams can be manipulated through Drupal by non-developers.  You can create a variety of ways to view this metadata and information related to the objects. Doing so requires knowledge of Drupal 8, but this essentially means that there are many ways to view the metadata and access the related objects in Islandora 8. 
```

File: docs/user-documentation/objects_to_resource_nodes.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+## Traditional objects become resource nodes
+
+Until now we have thought of objects in Islandora 8 in the conventional Islandora 7 way: A file loaded in the repository with associated derivatives. It can be helpful to move away from this when learning about Islandora 8. 
+
+One way to do this is to think of objects in Islandora 8 as _resource nodes_. While using the term “object” is traditional with Islandora it can get in the way of understanding the differences and possibilities in Islandora 8. Using the term “resource node” reflects the fact that files such as JPEGs, MP3s, .zip, etc. that get loaded through the data entry (formerly ingest) form are now fully integrated within Drupal and can take advantage of everything that represents. 
+
+The term resource node does not just refer to the individual files, however. The term reflects the new nature of objects in Islandora 8. A resource node encompasses multiple elements that all relate to each other, even if they are no longer directly connected like objects in Islandora 7. 
+
+The elements of a resource node are, typically:
```
Comment (librarychik) [inline docs/user-documentation/objects_to_resource_nodes.md]:
```suggestion
The typical elements of a resource node:
```

File: docs/user-documentation/objects_to_resource_nodes.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+-   Content types (one or more)
+-   Media files (the actual files of JPEGs, MP3s, .zip, etc.) that get loaded through the form
+-   Metadata fields entered in the data entry for, such as:
```
Comment (librarychik) [inline docs/user-documentation/objects_to_resource_nodes.md]:
```suggestion
-   Metadata fields submitted in the data entry, for example:
```

File: docs/user-documentation/objects_to_resource_nodes.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    -   A field denoting the 'type' of thing you're persisting (image, book, newspaper, etc...)
+    -   A field that creates the familiar collection hierarchy
+    -   Your custom fields you need to describe things the way you want to
```
Comment (librarychik) [inline docs/user-documentation/objects_to_resource_nodes.md]:
```suggestion
    -   Descriptive custom fields
```

File: docs/user-documentation/objects_to_resource_nodes.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+-   Derivative files (thumbnails, web-friendly service files, technical metadata, and more)
+
+It is possible to configure all elements of a resource node in Islandora 8. This fact allows full control over how one accesses the node and how they are displayed and discovered online. It also allows a repository to take full advantage of all third-party Drupal modules, themes, and distributions available.
```
Comment (librarychik) [inline docs/user-documentation/objects_to_resource_nodes.md]:
```suggestion
It is possible to configure all elements of a resource node in Islandora 8. This fact allows full control over how one accesses the node and how nodes are displayed and discovered online. It also allows a repository to take full advantage of all third-party Drupal modules, themes, and distributions available.  
```

File: docs/user-documentation/objects_to_resource_nodes.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+
+## Islandora 8: Empowering resource nodes through Drupal
+
+These resource nodes are what the librarian, student, archivist, technician, or general non-developer creates through the data entry form. It is a lot more than just the file (JPEG, .MP3, etc.). This user is creating the node and elements that make it possible to access, manage, and preserve the resource. Islandora 8 is all about empowering this access to repository resources. It makes it possible to touch and manipulate the file itself and the elements that make it a wholly preserved and shareable digital object. This shift is actually at the core of digital asset management and librarianship – making things accessible. Islandora 8 is about doing that through Drupal 8, through the web, making it that much more available, flexible, and viable for the future. It does this by making more elements of what was a traditional DAM object configurable and manageable by more users – _non-developers_. Work is ongoing by the community to increase the ease of use of the admin user interface to make this even more feasible.
```
Comment (librarychik) [inline docs/user-documentation/objects_to_resource_nodes.md]:
```suggestion
These resource nodes are what the librarian, student, archivist, technician, or general non-developer creates through the data entry form. It is a lot more than just the file (JPEG, .MP3, etc.). This user is creating the node and elements that make it possible to access, manage, and preserve the resource. Islandora 8 is all about empowering this access to repository resources. It makes it possible to touch and manipulate the file itself and the elements that make it a wholly preserved and shareable digital object. This shift is actually at the core of Digital Asset Management (DAM) and librarianship – making things accessible. Islandora 8 is about doing that through Drupal 8 and the web, making it that much more available, flexible, and viable for the future. It does this by making more elements of what was a traditional DAM object configurable and manageable by more users – _non-developers_. Work is ongoing by the community to increase the ease of use of the admin user interface to make this even more feasible. 
```

File: docs/user-documentation/objects_to_resource_nodes.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+Discovery attributes would include metadata fields, fields that define the collection hierarchy/structure, and any custom fields. They are any attributes of the resource node element that help manage how the resource node can be discovered through the interface and accessed by the public online. 
+
+For example, the following are discovery attributes:
+
+![screenshot of viewer and metadata](../assets/objects_to_resouce_nodes_attributes.jpg)
+
+### Concept: Resource node management attributes
+
+Management attributes of the resource node help to clarify and manage a resource nodes relationship within the repository and other content. Management attributes define if a node is a resource node, which collection it belongs to, the type of resource a node is (image, book, newspaper, etc.), and similar actions. When looking at Drupal functions, the following are examples of management attributes:
```
Comment (librarychik) [inline docs/user-documentation/objects_to_resource_nodes.md]:
```suggestion
Management attributes of the resource node help to clarify and manage its relationship within the repository and to other content. Management attributes define if a node is a resource node, which collection it belongs to, the type of resource a node is (image, book, newspaper, etc.), and similar actions. When looking at Drupal functions, the following are examples of management attributes: 
```

File: docs/user-documentation/objects_to_resource_nodes.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+For example, the following are discovery attributes:
+
+![screenshot of viewer and metadata](../assets/objects_to_resouce_nodes_attributes.jpg)
+
+### Concept: Resource node management attributes
+
+Management attributes of the resource node help to clarify and manage a resource nodes relationship within the repository and other content. Management attributes define if a node is a resource node, which collection it belongs to, the type of resource a node is (image, book, newspaper, etc.), and similar actions. When looking at Drupal functions, the following are examples of management attributes: 
+
+- Content types 
+    -   Default content types available are an article, basic page, repository item, video
```
Comment (librarychik) [inline docs/user-documentation/objects_to_resource_nodes.md]:
```suggestion
    -   Default content types available include article, basic page, repository item, video
```

File: docs/user-documentation/objects_to_resource_nodes.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    -   Able to add any content types that are needed to help define the resource node
```
Comment (librarychik) [inline docs/user-documentation/objects_to_resource_nodes.md]:
```suggestion
    -   Ability to add any content type needed to help define the resource node 
```

**Review by librarychik — CHANGES_REQUESTED** [review]
Made a number of minor suggested edits.  tagging @Islandora/8-x-committers

**Comment by manez** [comment]
Thank you @librarychik ! The new commits made that super easy 😄

**Comment by manez** [comment]
@Islandora/8-x-committers I think this one is ready to go in

**Comment by manez** [comment]
@seth-shaw-unlv or @whikloj can I impose? I need this one merged so I can delete my fork and start over on some different stuff 😅

**Review by seth-shaw-unlv — APPROVED** [review]
As per @manez.

**Comment by seth-shaw-unlv** [comment]
Ih, wait, this is a @manez pr... let me read through.

**Review by whikloj — APPROVED** [review]
This is all good, my only comment and it can happen later would be that a screenshot where the image viewer is not displaying an error is more polished.

**Comment by seth-shaw-unlv** [comment]
Thanks for covering me there, @whikloj! I'm out of the office this week and got distracted before I could finish the read-through.

**Pull request merged**